src/libraries/Common/src/System/Security/Cryptography/X509Certificates/Pkcs12LoaderLimits.cs

```diff
@@ -0,0 +1,350 @@
+                    PreserveUnknownAttributes = true,
+                });
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="Pkcs12LoaderLimits"/> class
+        ///   with default values.
+        /// </summary>
+        public Pkcs12LoaderLimits()
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="Pkcs12LoaderLimits"/> class
+        ///   by copying the values from another instance.
+        /// </summary>
+        /// <param name="copyFrom">The instance to copy the values from.</param>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="copyFrom"/> is <see langword="null" />.
+        /// </exception>
+        public Pkcs12LoaderLimits(Pkcs12LoaderLimits copyFrom)
+        {
+            ArgumentNullException.ThrowIfNull(copyFrom);
+
+            // Do not copy _isReadOnly.
+
+            _macIterationLimit = copyFrom._macIterationLimit;
+            _individualKdfIterationLimit = copyFrom._individualKdfIterationLimit;
+            _totalKdfIterationLimit = copyFrom._totalKdfIterationLimit;
+            _maxKeys = copyFrom._maxKeys;
+            _maxCertificates = copyFrom._maxCertificates;
+            _preserveStorageProvider = copyFrom._preserveStorageProvider;
+            _preserveKeyName = copyFrom._preserveKeyName;
+            _preserveCertificateAlias = copyFrom._preserveCertificateAlias;
+            _preserveUnknownAttributes = copyFrom._preserveUnknownAttributes;
+            _ignorePrivateKeys = copyFrom._ignorePrivateKeys;
+            _ignoreEncryptedAuthSafes = copyFrom._ignoreEncryptedAuthSafes;
+        }
+
+        /// <summary>
+        ///   Gets a value indicating whether the instance is read-only.
+        /// </summary>
+        /// <value>
+        ///   <see langword="true" /> if the instance is read-only; otherwise, <see langword="false" />.
+        /// </value>
+        public bool IsReadOnly => _isReadOnly;
+
+        /// <summary>
+        ///   Makes the <see cref="Pkcs12LoaderLimits"/> instance read-only.
+        /// </summary>
+        public void MakeReadOnly()
+        {
+            _isReadOnly = true;
+        }
+
+        private static Pkcs12LoaderLimits MakeReadOnly(Pkcs12LoaderLimits limits)
+        {
+            limits.MakeReadOnly();
+            return limits;
+        }
+
+        private void CheckReadOnly()
+        {
+            if (_isReadOnly)
+            {
+                throw new InvalidOperationException(SR.Cryptography_X509_PKCS12_LimitsReadOnly);
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets the iteration limit for the MAC calculation.
+        /// </summary>
+        /// <value>The iteration limit for the MAC calculation.</value>
+        public int? MacIterationLimit
+        {
+            get => _macIterationLimit;
+            set
+            {
+                if (value < 0)
```

Comment:
What does a MAC iteration limit of zero mean? “Do not permit a MAC?”

Reply:
Yeah, effectively.